r2dbc-mysql/src/main/java/SimpleStatement.kt

```diff
@@ -62,7 +95,7 @@ class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, pr
             } else {
                 jasyncConnection.sendQuery(this.sql)
             }
-            r.toMono().map { JaysncResult(it.rows, it.rowsAffected) }
+            r.toMono()
```

Comment:
I think the way to go (to save the additional roundtrip) is to cast the result here to `MySQLQueryResult` and use the `lastInsertId` property.